src/nimble.nim

```diff
@@ -673,6 +673,28 @@ proc execBackend(options: Options) =
         [backend, args, bin], showOutput = true)
   display("Success:", "Execution finished", Success, HighPriority)
 
+proc tempOutArg(file: string): string =
+  ## Returns the `--out` argument to pass to the compiler, using a temp file
+  let (_, name, _) = splitFile(file)
+  let dir = getNimbleTempDir() / "tests"
+  createDir(dir)
+  result = "--out:" & (dir / name)
+
+proc test(options: Options) =
+  ## Executes all tests
+  var files = toSeq(walkDir(getCurrentDir() / "tests"))
+  files.sort do (a, b: auto) -> int:
+    result = cmp(a.path, b.path)
+
+  for file in files:
+    if file.path.endsWith(".nim") and file.kind in { pcFile, pcLinkToFile }:
+      var optsCopy = options
+      optsCopy.action.file = file.path
+      optsCopy.action.backend = "c -r"
+      optsCopy.action.compileOptions.add("--path:.")
+      optsCopy.action.compileOptions.add(file.path.tempOutArg)
+      execBackend(optsCopy)
+
```

Comment:
I'm not sure whether ``execBackend`` handles this but we need to be sure to exit with an error exit code.

Also, test fail/pass messages would be nice.